### PR TITLE
DOCS-8728 LLM Observability API Reference Edit

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Adds a note on the IO message section explaining how Datadog infers the IO value from the message.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/llm_observability/setup/api.md [hunk 1]
```diff
@@ -144,7 +144,13 @@ If the request is successful, the API responds with a 202 network code and an em
 | value   | string | Input or output value. This should be used for all spans except for LLM spans. |
 | messages| [[Message](#message)] | List of messages. This should only be used for LLM spans. |
 
+**Note**: When only `input.messages` is set for an LLM span, Datadog infers `input.value` from `input.messages` and uses the following inference logic:
+
+1. If a message with `role=user` exists, the content of the last message is used as `input.value`. 
+1. If a `user` role message is not present, the content is inferred by concatenating all messages that do not have the `role=system` or `role=function`.
```
Comment (lievan) [inline content/en/llm_observability/setup/api.md:150]: Just recently updated the logic to be
- "input.value is inferred by concatenating the `content` field of all messages"
(so there's no role exclusion)

Comment (alai97) [inline content/en/llm_observability/setup/api.md:150]:
```suggestion
**Note**: When only `input.messages` is set for an LLM span, Datadog infers `input.value` by concatenating the content fields of all messages, regardless of their roles.
```

File: content/en/llm_observability/setup/api.md [hunk 2]
```diff
@@ -144,7 +144,14 @@ If the request is successful, the API responds with a 202 network code and an em
 | value   | string | Input or output value. This should be used for all spans except for LLM spans. |
 | messages| [[Message](#message)] | List of messages. This should only be used for LLM spans. |
 
+
+**Note**: When only `input.messages` is set for an LLM span, Datadog infers `input.value` from `input.messages` and uses the following inference logic:
+
+1. If a message with `role=user` exists, the content of the last message is used as `input.value`.
+1. If a `user` role message is not present, `input.value` is inferred by concatenating the content fields of all messages, regardless of their roles.
```
Comment (lievan) [inline content/en/llm_observability/setup/api.md:151]: is this supposed to be "2."?

**Review by lievan — APPROVED** [review]

File: content/en/llm_observability/setup/api.md (hunk 2 above)
Comment (alai97) [reply content/en/llm_observability/setup/api.md:151]: This is fine actually, our doc builds it as a numbered list of instructions :D

**Review by brett0000FF — APPROVED** [review]
🚀

**Pull request merged**